contracts/eosio.sudo/eosio.sudo.cpp

```diff
@@ -19,6 +19,9 @@ void sudo::exec() {
    constexpr size_t max_stack_buffer_size = 512;
    size_t size = action_data_size();
    char* buffer = (char*)( max_stack_buffer_size < size ? malloc(size) : alloca(size) );
+   struct { 
+      struct S { ~S()  { if (buf) free(buf); } char *buf; } s; 
+   } __freemem{max_stack_buffer_size<size?buffer:0};  
    read_action_data( buffer, size );
```

Comment:
Not needed, this will be freed at end of action.
Please submit changes to this contract to the https://github.com/EOSIO/eosio.contracts repo.